File: CPP/bubblesort.cpp
#include <iostream>
using namespace std;

void bubblesort(int n, int arr[])
{
  for(int i = 0; i < n-1; i++){
    for(int j = n-1; j > i; j--){
      if(arr[j-1] > arr[j]){
        int temp = arr[j-1];
        arr[j-1] = arr[j];
        arr[j] = temp;
      }
    }
  }
}

int main()
{
  int n = 5;
  int arr[] = {2, 1, 4, 3, 5};

  for(int i = 0; i < n; i++){
    cout << arr[i] << " ";
  }
  cout << endl;

  bubblesort(n, arr);
  
  for(int i = 0; i < n; i++){
    cout << arr[i] << " ";
  }
  cout << endl;

  return 0;
}

File: Swift/bubble_sort.swift
// Bubble sort algorithm 

// Input elements: array that implement the comparable protocol

public func bubbleSort<T> (_ elements: [T]) -> [T] where T: Comparable {
  return bubbleSort(elements, <)
}

public func bubbleSort<T> (_ elements: [T], _ comparison: (T,T) -> Bool) -> [T]  {
  var array = elements
  
  for i in 0..<array.count {
    for k in 1..<array.count-i {
      if comparison(array[k], array[k-1]) {
        let temp = array[k-1]
        array[k-1] = array[k]
        array[k] = temp
      }
    }
  }
  
  return array
}
